proposed/2022/dotnet-nuget-config-spec.md

```diff
@@ -0,0 +1,298 @@
+# NuGet configuration CLI for dotnet.exe
+
+- Author: [Heng Liu](https://github.com/heng-liu)
+- GitHub Issue [8420](https://github.com/NuGet/Home/issues/8420)
+
+## Problem Background
+
+Currently, there is no NuGet configuration CLI for dotnet.exe. It's inconvenient for NuGet users to know about the NuGet configuration file locations and figure out where is the merged configuration coming from. 
+
+## Who are the customers
+
+This feature is for dotnet.exe users.
+
+## Goals
+Design and implement `dotnet nuget config` command.
+
+## Non-Goals
+Design and implement `dotnet nuget config` command with commands other than `list`, e.g. add/update/delete
+
+## Solution
+The following command will be implemented in the `dotnet.exe` CLI.
+
+### `dotnet nuget config`
+
+#### Commands
+
+- List
+
+Lists all the NuGet configuration settings. 
+
+This command will list merged NuGet configuration settings from one or multiple NuGet configuration files that will be applied, when invoking NuGet command from the current working directory path. 
+
+#### Arguments
+
+- WORKING_DIRECTORY
+
+Run this command as if working directory is set to the specified directory.
+
+If the specified `WORKING_DIRECTORY` doesn't exist, an error is displayed indicating the `WORKING_DIRECTORY` doesn't exist.
+
+> [!Note]
+> If `WORKING_DIRECTORY` (or its parent directories) is not accessible, the command will ignore any NuGet configuration files under those directories without any warning/error. This is aligned with other NuGet commands.
```

Comment:
This argument feels odd to me - it's not used in any of the other proposed commands (instead explicit `--config-file` options are provided), so why does this concept get first-class treatment here?

Looking at the other commands (set, unset) to me `list` implies listing configuration values (parallel to setting/unsetting them), _not_ locating and showing the entire xml config file.  This command as described feels like a `dotnet nuget config paths` kind of command as a result.

Reply:
The verb `get` is missing here, and it seems like a natural analogue for users that are scripting. Are scripting use cases a motivation here?

Reply:
dotnet store command has -w|--working-dir <WORKING_DIRECTORY> option https://learn.microsoft.com/dotnet/core/tools/dotnet-store#optional-options

Reply:
Thanks for reviewing!
The option is renamed to `-w|--working-dir <WORKING_DIRECTORY>`. Thanks @kartheekp-ms!
The `get` is added. Thanks @baronfel !

Reply:
> Thanks for reviewing! The option is renamed to `-w|--working-dir <WORKING_DIRECTORY>`. Thanks @kartheekp-ms! The `get` is added. Thanks @baronfel !

I don't think that was the outcome in the meeting. 

When we brought this up for discussion, Daniel pointed out that other commands such as `dotnet build`, `dotnet nuget sources` etc all use a directory or a project file as the first argument. 

I was under the impression that no one disagreed with that? 

I guess there's a recording so we can always figure it out from there :)